partiql-spi/src/main/java/org/partiql/spi/value/Datum.java

```diff
@@ -596,14 +603,82 @@ static Datum string(@NotNull String value) {
         return new DatumString(value, PType.string());
     }
 
+    /**
+     *
+     * @param value the string to place in the varchar
+     * @return a varchar value with a default length of 255
+     */
+    @NotNull
+    static Datum varchar(@NotNull String value) {
+        return varchar(value, 255);
+    }
+
+    /**
+     *
+     * @param value the string to place in the varchar
+     * @return a varchar value
+     * TODO: Error or coerce here? Right now coerce, though I think this should likely error.
+     */
+    @NotNull
+    static Datum varchar(@NotNull String value, int length) {
+        String newValue;
+        if (length <= 0) {
+            throw new DataException("VARCHAR of length " + length + " not allowed.");
+        }
+        if (value.length() < length) {
+            newValue = String.format("%-" + length + "." + length + "s", value);
+        } else if (value.length() == length) {
+            newValue = value;
+        } else {
+            newValue = value.substring(0, length);
+        }
+        return new DatumString(newValue, PType.varchar(length));
+    }
+
+    /**
+     *
+     * @param value the string to place in the char
+     * @return a char value with a default length of 255
+     */
+    @NotNull
+    static Datum character(@NotNull String value) {
+        return character(value, 255);
+    }
+
+    /**
+     *
+     * @param value the string to place in the char
+     * @return a char value
+     */
+    @NotNull
+    static Datum character(@NotNull String value, int length) {
+        String newValue;
+        if (length <= 0) {
+            throw new DataException("CHAR of length " + length + " not allowed.");
+        }
+        if (value.length() < length) {
+            newValue = String.format("%-" + length + "." + length + "s", value);
+        } else if (value.length() == length) {
+            newValue = value;
+        } else {
+            newValue = value.substring(0, length);
+        }
+        return new DatumString(newValue, PType.varchar(length));
```

Comment:
should be `character`?

```suggestion
        return new DatumString(newValue, PType.character(length));
```